backend/src/tasks/lookingGlass.ts

```diff
@@ -0,0 +1,316 @@
+  const resource =
+    'https://delta.lookingglasscyber.com/api/v1/workspaces/' +
+    process.env.LG_WORKSPACE_NAME +
+    '/collections';
+  return (await got(resource, {
+    headers: { Authorization: 'Bearer ' + process.env.LG_API_KEY }
+  }).json()) as LGCollectionsResponse[];
+}
+
+function validateIPAddress(ipAddress) {
+  return /\b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b/.test(ipAddress);
+}
+
+/**
+ * Gets threat information from the LookingGlass API
+ * for the collection with the given ID.
+ */
+async function getThreatInfo(collectionID) {
+  const resource = 'https://delta.lookingglasscyber.com/api/graph/query';
+  const modifier = {
+    period: 'all',
+    query: [
+      'and',
+      ['=', 'type', ['associated-with']],
+      ['or', ['=', 'right.type', 'threat'], ['=', 'left.type', 'ipv4']],
+      ['=', 'left.collectionIds', collectionID]
+    ],
+    fields: ['firstSeen', 'lastSeen', 'sources', 'right', 'left'],
+    limit: 100000,
+    workspaceIds: []
+  };
+
+  return (got
+    .post(resource, {
+      json: modifier,
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: 'Bearer ' + process.env.LG_API_KEY
+      }
+    })
+    .json() as unknown) as LGThreatResponse;
+}
+/**
+ * Save domains from LookingGlass API response, and retrieve
+ * domains for the given organization.
+ */
+async function saveAndPullDomains(
+  response: LGThreatResponse,
+  organizationId: string,
+  scanId: string
+) {
+  const domains: Domain[] = [];
+  for (const l of response.results) {
+    if (validateIPAddress(l.left.name)) {
+      // If l.left.name is an IP address, the domain has no associated
+      // domain name, so mark it as an ip-only domain.
+      const currentDomain = plainToClass(Domain, {
+        name: l.left.name,
+        ip: l.left.name,
+        organization: { id: organizationId },
+        fromRootDomain: '',
+        ipOnly: true,
+        discoveredBy: { id: scanId }
+      });
+      domains.push(currentDomain);
+    } else {
+      const currentDomain = plainToClass(Domain, {
+        name: l.left.name,
+        ip: '',
+        organization: { id: organizationId },
+        fromRootDomain: '',
+        ipOnly: false,
+        discoveredBy: { id: scanId }
+      });
+      domains.push(currentDomain);
+    }
+  }
+  await saveDomainsToDb(domains);
+
+  const pulledDomains = Domain.createQueryBuilder('domain')
+    .leftJoinAndSelect('domain.organization', 'organization')
+    .andWhere('ip IS NOT NULL')
+    .andWhere('domain.organization=:org', { org: organizationId });
+
+  return pulledDomains.getMany();
+}
+
+export const handler = async (commandOptions: CommandOptions) => {
+  const { organizationId, organizationName, scanId } = commandOptions;
+
+  console.log('Running lookingGlass on organization', organizationName);
+
+  const collections: LGCollectionsResponse[] = await getCollectionForCurrentWorkspace();
+  const ipsAndDomains: string[] = [];
```

Comment:
We should use a Set instead of an array for efficiency (since we're just adding things / checking if they exist):

```suggestion
  const ipsAndDomains: Set<string> = new Set();
```